src/layers/embeddings.jl

```diff
@@ -1,27 +1,35 @@
 """
-    PatchEmbedding(patch_size)
-    PatchEmbedding(patch_height, patch_width)
+    PatchEmbedding(imsize::NTuple{2} = (224, 224); inchannels = 3, patch_size = (16, 16), 
+                   embedplanes = 768, norm_layer = identity, flatten = true)
 
-Patch embedding layer used by many vision transformer-like models to split the input image into patches.
+Patch embedding layer used by many vision transformer-like models to split the input image into 
+patches.
+
+# Arguments:
+- `imsize`: the size of the input image
+- `inchannels`: the number of channels in the input image
+- `patch_size`: the size of the patches
+- `embedplanes`: the number of channels in the embedding
+- `norm_layer`: the normalization layer
+- `flatten`: whether to flatten the input after the embedding
 """
-struct PatchEmbedding
-  patch_height::Int
-  patch_width::Int
-end
+function PatchEmbedding(imsize::NTuple{2} = (224, 224); inchannels = 3, patch_size = (16, 16), 
+                        embedplanes = 768, norm_layer = identity, flatten = true)
 
-PatchEmbedding(patch_size) = PatchEmbedding(patch_size, patch_size)
+  im_height, im_width = imsize
+  patch_height, patch_width = patch_size
 
-function (p::PatchEmbedding)(x)
-  h, w, c, n = size(x)
-  hp, wp = h ÷ p.patch_height, w ÷ p.patch_width
-  xpatch = reshape(x, hp, p.patch_height, wp, p.patch_width, c, n)
+  norm_layer = typeof(norm_layer) != typeof(identity) ? norm_layer(embedplanes) : identity
```

Comment:
I would avoid this and make the keyword argument default to `planes -> identity`. And specify in the docstring that `norm_layer` should be a single arg constructor.